cupy/core/_routines_indexing.pyx

```diff
@@ -62,18 +72,18 @@ cdef tuple _ndarray_nonzero(ndarray self):
         count_nonzero = int(scan_index[-1])  # synchronize!
     ndim = max(<int>self._shape.size(), 1)
     if count_nonzero == 0:
-        return (ndarray((0,), dtype=dtype),) * ndim
+        return ndarray((0, ndim), dtype=dtype)
 
     if ndim <= 1:
-        dst = ndarray((count_nonzero,), dtype=dtype)
+        dst = ndarray((count_nonzero, 1), dtype=dtype)
         _nonzero_kernel_1d(nonzero, scan_index, dst)
```

Comment:
There seems to be a special case kernel for the 1-D case. Maybe we don't need this any more and can drop it? Possibly related to the dimensionality differences between CuPy and NumPy identified.